MdeModulePkg/Core/Dxe/Hand/Handle.c

```diff
@@ -59,23 +114,16 @@ CoreValidateHandle (
   IN  EFI_HANDLE  UserHandle
   )
 {
-  IHANDLE     *Handle;
-  LIST_ENTRY  *Link;
+  ORDERED_COLLECTION_ENTRY  *Entry;
 
   if (UserHandle == NULL) {
     return EFI_INVALID_PARAMETER;
   }
 
   ASSERT_LOCKED (&gProtocolDatabaseLock);
 
-  for (Link = gHandleList.BackLink; Link != &gHandleList; Link = Link->BackLink) {
-    Handle = CR (Link, IHANDLE, AllHandles, EFI_HANDLE_SIGNATURE);
-    if (Handle == (IHANDLE *)UserHandle) {
-      return EFI_SUCCESS;
-    }
-  }
-
-  return EFI_INVALID_PARAMETER;
+  Entry = OrderedCollectionFind (gOrderedHandleList, UserHandle);
+  return Entry ? EFI_SUCCESS : EFI_INVALID_PARAMETER;

```

Comment:
if (Entry == NULL) {
  return EFI_INVALID_PARAMETER;
}
return EFI_SUCCESS;

Reply:
Thanks Ray, changed as following to align with original code logical

  if (Entry != NULL) {
    return EFI_SUCCESS;
  }

  return EFI_INVALID_PARAMETER;